gadget-container/entrypoint.sh

```diff
@@ -39,20 +39,33 @@ ARGS=k8s
 FLATCAR_EDGE=0
 if grep -q '^ID=flatcar$' /host/etc/os-release > /dev/null ; then
   if grep -q '^GROUP=edge$' /host/etc/flatcar/update.conf > /dev/null ; then
+    echo "Flatcar Edge detected."
     FLATCAR_EDGE=1
+
+    CGROUP_V1_PATH=$(cat /proc/1/cgroup |grep ^1:|cut -d: -f3)
+    CGROUP_V2_PATH=$(cat /proc/1/cgroup |grep ^0:|cut -d: -f3)
+    if [ $CGROUP_V1_PATH != $CGROUP_V2_PATH ] ; then
+      echo "cgroup-v2 is not correctly enabled on Kubernetes pods" >&2
+      exit 1
+    fi
   fi
 fi
 
-if [ "$FLATCAR_EDGE" = 1 ] ; then
-  echo "Flatcar Edge detected."
-  echo "Installing scripts on host..."
-
-  CGROUP_V1_PATH=$(cat /proc/1/cgroup |grep ^1:|cut -d: -f3)
-  CGROUP_V2_PATH=$(cat /proc/1/cgroup |grep ^0:|cut -d: -f3)
-  if [ $CGROUP_V1_PATH != $CGROUP_V2_PATH ] ; then
-    echo "cgroup-v2 is not correctly enabled on Kubernetes pods" >&2
-    exit 1
+if [ "$INSPEKTOR_GADGET_OPTION_RUNC_HOOKS" = "ldpreload" ] ; then
+  echo "Installing ld.so.preload with runchooks.so for OCI hooks"
+  mkdir -p /host/opt/runchooks/
+  cp /opt/runchooks/runchooks.so /host/opt/runchooks/
+  cp /opt/runchooks/add-hooks.jq /host/opt/runchooks/
+  touch /host/etc/ld.so.preload
+  if grep -q ^/opt/runchooks/runchooks.so$ /host/etc/ld.so.preload > /dev/null ; then
+    echo "runchooks.so already setup in /etc/ld.so.preload"
+  else
+    echo "/opt/runchooks/runchooks.so" >> /host/etc/ld.so.preload
   fi
+fi
+
+if [ "$FLATCAR_EDGE" = 1 ] || [ "$INSPEKTOR_GADGET_OPTION_RUNC_HOOKS" = "ldpreload" ] ; then
```

Comment:
I don't understand this line. Shouldn't it compare if `RUNC_HOOKS` is `auto`? Why to install `runc-hook-prestart.sh runc-hook-poststop.sh` if they are not needed in the ldpreload case?

Reply:
Ah got it, I thought add-hooks.jq was invoking `ocihookgadget` directly.

Reply:
I did that initially, but then `ocihookgadget` could return errors (e.g. if the gadgettracermanager is no longer running because the gadget pod has been stopped) and that was breaking all new pods.
Now, the scripts like `gadget-container/ocihookgadget/runc-hook-prestart.sh` have this protection:
```
pidof gadgettracermanager > /dev/null || exit 0
```
So new pods don't crash if the gadget pod is no longer running.

Alternatively, we could get the grpc client to avoid returning any errors if the grpc server is not running.